### ISSUE TITLE
Deleted resource but didn't deploy

### ISSUE THREAD

**Issue by iHearty** [jenkinsci/kubernetes-cd-plugin#100]
Env:
   jenkins: 2.176.2
  kubernetes-cd: 2.1.0
  k8s: 1.15

I configure plugin with "Delete Resources", after task run, when i check the result in k8s dashboard, "Deployment" was deleted, but not recreate, "Pod" was not deleted.

```
# deployment.yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  namespace: spot-dev
  name: spot-base-data
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: spot-base-data
    spec:
      nodeSelector:
        space: spot-dev
      containers:
      - image: <DOCKER REGISTRY>/dev/spot-base-data:dev
        imagePullPolicy: Always
        name: spot-base-data
        env:
          - name: JAVA_OPTS
            value: "-Dfile.encoding=UTF8  -Duser.timezone=GMT+08"
          - name: "kubernetes.namespace"
            value: "spot-dev"
        volumeMounts:
        - mountPath: /app
          name: app-volume
        - mountPath: /applogs
          name: log-volume
        - name: tz-config
          mountPath: /etc/localtime
      volumes:
        - name: app-volume
          emptyDir: {}
        - name: log-volume
          emptyDir: {}
        - name: tz-config
          hostPath:
            path: /usr/share/zoneinfo/Asia/Shanghai
      imagePullSecrets:
      - name: dockersecret
```

jenkins build output:

```
Starting Kubernetes deployment
Loading configuration: /apps/jenkins/workspace/spot-base-data/build/k8s/deployment.yaml
Deleted class ExtensionsV1beta1Deployment {
    apiVersion: extensions/v1beta1
    kind: Deployment
    metadata: class V1ObjectMeta {
        annotations: null
        clusterName: null
        creationTimestamp: null
        deletionGracePeriodSeconds: null
        deletionTimestamp: null
        finalizers: null
        generateName: null
        generation: null
        initializers: null
        labels: null
        name: spot-base-data
        namespace: spot-dev
        ownerReferences: null
        resourceVersion: null
        selfLink: null
        uid: null
    }
    spec: class ExtensionsV1beta1DeploymentSpec {
        minReadySeconds: null
        paused: null
        progressDeadlineSeconds: null
        replicas: 1
        revisionHistoryLimit: null
        rollbackTo: null
        selector: null
        strategy: null
        template: class V1PodTemplateSpec {
            metadata: class V1ObjectMeta {
                annotations: null
                clusterName: null
                creationTimestamp: null
                deletionGracePeriodSeconds: null
                deletionTimestamp: null
                finalizers: null
                generateName: null
                generation: null
                initializers: null
                labels: {app=spot-base-data}
                name: null
                namespace: null
                ownerReferences: null
                resourceVersion: null
                selfLink: null
                uid: null
            }
            spec: class V1PodSpec {
                activeDeadlineSeconds: null
                affinity: null
                automountServiceAccountToken: null
                containers: [class V1Container {
                    args: null
                    command: null
                    env: [class V1EnvVar {
                        name: JAVA_OPTS
                        value: -Dfile.encoding=UTF8  -Duser.timezone=GMT+08
                        valueFrom: null
                    }, class V1EnvVar {
                        name: kubernetes.namespace
                        value: spot-dev
                        valueFrom: null
                    }]
                    envFrom: null
                    image: 10.108.0.13x/dev/spot-base-data:dev
                    imagePullPolicy: Always
                    lifecycle: null
                    livenessProbe: null
                    name: spot-base-data
                    ports: null
                    readinessProbe: null
                    resources: null
                    securityContext: null
                    stdin: null
                    stdinOnce: null
                    terminationMessagePath: null
                    terminationMessagePolicy: null
                    tty: null
                    volumeDevices: null
                    volumeMounts: [class V1VolumeMount {
                        mountPath: /app
                        mountPropagation: null
                        name: app-volume
                        readOnly: null
                        subPath: null
                    }, class V1VolumeMount {
                        mountPath: /applogs
                        mountPropagation: null
                        name: log-volume
                        readOnly: null
                        subPath: null
                    }, class V1VolumeMount {
                        mountPath: /etc/localtime
                        mountPropagation: null
                        name: tz-config
                        readOnly: null
                        subPath: null
                    }]
                    workingDir: null
                }]
                dnsConfig: null
                dnsPolicy: null
                enableServiceLinks: null
                hostAliases: null
                hostIPC: null
                hostNetwork: null
                hostPID: null
                hostname: null
                imagePullSecrets: [class V1LocalObjectReference {
                    name: dockersecret
                }]
                initContainers: null
                nodeName: null
                nodeSelector: {space=spot-dev}
                priority: null
                priorityClassName: null
                readinessGates: null
                restartPolicy: null
                runtimeClassName: null
                schedulerName: null
                securityContext: null
                serviceAccount: null
                serviceAccountName: null
                shareProcessNamespace: null
                subdomain: null
                terminationGracePeriodSeconds: null
                tolerations: null
                volumes: [class V1Volume {
                    awsElasticBlockStore: null
                    azureDisk: null
                    azureFile: null
                    cephfs: null
                    cinder: null
                    configMap: null
                    downwardAPI: null
                    emptyDir: class V1EmptyDirVolumeSource {
                        medium: null
                        sizeLimit: null
                    }
                    fc: null
                    flexVolume: null
                    flocker: null
                    gcePersistentDisk: null
                    gitRepo: null
                    glusterfs: null
                    hostPath: null
                    iscsi: null
                    name: app-volume
                    nfs: null
                    persistentVolumeClaim: null
                    photonPersistentDisk: null
                    portworxVolume: null
                    projected: null
                    quobyte: null
                    rbd: null
                    scaleIO: null
                    secret: null
                    storageos: null
                    vsphereVolume: null
                }, class V1Volume {
                    awsElasticBlockStore: null
                    azureDisk: null
                    azureFile: null
                    cephfs: null
                    cinder: null
                    configMap: null
                    downwardAPI: null
                    emptyDir: class V1EmptyDirVolumeSource {
                        medium: null
                        sizeLimit: null
                    }
                    fc: null
                    flexVolume: null
                    flocker: null
                    gcePersistentDisk: null
                    gitRepo: null
                    glusterfs: null
                    hostPath: null
                    iscsi: null
                    name: log-volume
                    nfs: null
                    persistentVolumeClaim: null
                    photonPersistentDisk: null
                    portworxVolume: null
                    projected: null
                    quobyte: null
                    rbd: null
                    scaleIO: null
                    secret: null
                    storageos: null
                    vsphereVolume: null
                }, class V1Volume {
                    awsElasticBlockStore: null
                    azureDisk: null
                    azureFile: null
                    cephfs: null
                    cinder: null
                    configMap: null
                    downwardAPI: null
                    emptyDir: null
                    fc: null
                    flexVolume: null
                    flocker: null
                    gcePersistentDisk: null
                    gitRepo: null
                    glusterfs: null
                    hostPath: class V1HostPathVolumeSource {
                        path: /usr/share/zoneinfo/Asia/Shanghai
                        type: null
                    }
                    iscsi: null
                    name: tz-config
                    nfs: null
                    persistentVolumeClaim: null
                    photonPersistentDisk: null
                    portworxVolume: null
                    projected: null
                    quobyte: null
                    rbd: null
                    scaleIO: null
                    secret: null
                    storageos: null
                    vsphereVolume: null
                }]
            }
        }
    }
    status: null
}: null
Finished Kubernetes deployment
```
![1566870279(1)](https://user-images.githubusercontent.com/10851527/63734598-7b997f80-c8af-11e9-87a6-8b15be0a5766.jpg)

**Comment by gavinfish** [comment]
@iHearty I do not think the plugin should recreate a deployment after its deletion.
@runzexia We should take care of its pods after a deployment/rc being deleted.

**Comment by iHearty** [comment]
@gavinfish Does "Delete Resources" mean that this step just only delete, if i need to deploy resources, i need create another step with "Delete Resources" checkbox unchecked?
Thanks for you reply very much.

**Comment by gavinfish** [comment]
@iHearty For current design, I have to say yes. It will only delete resources which are no longer needed.

**Comment by runzexia** [comment]
The pod is not cleared. It may be a problem with the deletion policy. I will try it in my environment.

**Comment by iHearty** [comment]
@gavinfish  Thank you very much for your reply and patience.
I used two-step configuration tested it, one step with "Delete Resources" checked, another step with "Delete Resources" unchecked. and i got the below result, "Deployment" and "Pod" have deffirent creation time.
![image](https://user-images.githubusercontent.com/10851527/63735921-febcd480-c8b3-11e9-9027-df52747907bf.png)

**Comment by gavinfish** [comment]
Released as 2.1.1 .
